File: Samples/android-Crashlytics/app/src/main/java/com/example/user/android_crashlytics/MainActivity.java
package com.example.user.android_crashlytics;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.crashlytics.android.Crashlytics;
import io.fabric.sdk.android.Fabric;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        //Initialize the crashlytics lib from fabric
        Fabric.with(this, new Crashlytics());
        setContentView(R.layout.activity_main);
    }


    //This button click is forcing the app to crash.
    //Check your email or crashlytics console for more details about the crash.
    public void forceCrash(View view) {
        throw new RuntimeException("This is a crash");
    }


}

File: Samples(kotlin)/kotlin-basicconcepts/app/src/main/java/com/example/user/kotlin_basicconcepts/MainActivity.kt
package com.example.user.kotlin_basicconcepts

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import com.example.user.kotlin_basicconcepts.ClassesPropertiesAndInterfaces.MediaAdapter
import com.example.user.kotlin_basicconcepts.ClassesPropertiesAndInterfaces.MediaItem

class MainActivity : AppCompatActivity() {



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        toast("This is the default text")//function call to toast method
        val textView: TextView = findViewById(R.id.textView)//bind the textview
        textView.text = "Hello Kotlin"//assign a value to the text using properties not setText()

        /*
        Bind the views to show the toast from an EditText from the button click
        */
        val editText: EditText = findViewById(R.id.message)

        var button: Button = findViewById(R.id.button)

        /*
        For single method interfaces, we can use lambdas to put the block on code in the
        method definition.

        String template
        We are also using a string template in the method call us concatenate the strings
        before and after the EditText value
        */
        button.setOnClickListener { toast("Hello ${editText.text}, how are you doing") }


        //bind the RecyclerView
        val recyclerView: RecyclerView = findViewById(R.id.recycler)
        recyclerView.layoutManager = GridLayoutManager(this,2)
        recyclerView.adapter = MediaAdapter(ArrayList())


    }

    /*
    Toast is a function that is displaying a text using the Toast class
    */
    private fun toast(s: String) {
        Toast.makeText(this, s, Toast.LENGTH_SHORT).show()
    }



}
